controller/internal/routes/enroll_router.go

```diff
@@ -105,51 +128,44 @@ func (ro *EnrollRouter) enrollHandler(ae *env.AppEnv, rc *response.RequestContex
 	err := enrollContext.FillFromHttpRequest(rc.Request)
 
 	if err != nil {
-		rc.RequestResponder.RespondWithError(err)
+		rc.RespondWithError(err)
 		return
 	}
 
 	result, err := ae.Handlers.Enrollment.Enroll(enrollContext)
 
 	if err != nil {
-		rc.RequestResponder.RespondWithError(err)
+		rc.RespondWithError(err)
 		return
 	}
 
 	if result == nil {
-		rc.RequestResponder.RespondWithUnauthorizedError(rc)
+		rc.RespondWithApiError(apierror.NewUnauthorized())
 		return
 	}
 
-	err = ro.Respond(rc, result)
-
-	if err != nil {
-		pfxlog.Logger().WithError(err).Error("error attempting to respond with enrollment response")
-	}
-}
-
-func (ro *EnrollRouter) Respond(rc *response.RequestContext, enrollmentResult *model.EnrollmentResult) error {
-	rc.ResponseWriter.Header().Add("content-type", enrollmentResult.ContentType)
-
-	contentType := strings.Split(enrollmentResult.ContentType, ";")
-
-	if contentType[0] == "application/json" {
-		data := map[string]interface{}{}
-		err := json.Unmarshal(enrollmentResult.Content, &data)
-		if err != nil {
-			return err
+	//prefer json producer for non ott methods (backwards compat for ott)
+	explicitJsonAccept := false
+	if accept := rc.Request.Header.Values("accept"); len(accept) == 0 {
+		explicitJsonAccept = false //no headers specified
+	} else {
+		for _, val := range accept {
+			if strings.Split(val, ";")[0] == "application/json" {
```

Comment:
application/json is guaranteed to be at position 0? i don't know what `accept` will look like here so this is probably just fine again - just pointing it out to make sure

Reply:
This is another work-around for a larger issue - the same one I mentioned in another one of your comments dealing w/ `TextContent` in `EnrollmentResults`. 

This is essentially logic determines if we can use JSON as the response `content-type` at all and if we can prefer it. It is a concept that the `go-swagger` code doesn't attempt to solve for us - nor should it.

As for the format, it is specified by RFC 1049: `Content-Type:= type [";" ver-num [";" 1#resource-ref]] [comment]`